guides/OT_INSTRUMENTATION.md

```diff
@@ -0,0 +1,231 @@
+defer func() {
+err = errors.Join(err, otelShutdown(context.Background()))
+}()
+```
+NB: if this isn't done any calls to the otel service will fail silently. If you find that traces are not coming through, ensure this code is getting called.
+
+
+## Instrumenting http handlers
+There are a wide range of different facilities for instrumenting http calls. The simplest (taken here from dp-frontend-router) simply wraps a http handler (or Gorillamux router) with an opentelemetry handler:
+```
+httpHandler := router.New(routerConfig)
+otelHandler := otelhttp.NewHandler(httpHandler,"/")
+
+log.Info(ctx, "Starting server", log.Data{"config": cfg})
+
+s := &http.Server{
+Handler: otelHandler,
+ReadTimeout: cfg.ProxyTimeout,
+WriteTimeout: cfg.ProxyTimeout,
+IdleTimeout: 120 * time.Second,
+}
+```
+
+
+The following ensures that the http route name is included in all subsequent traces:
+```
+func routes(router *mux.Router, hc *healthcheck.HealthCheck) *RendererAPI {
+    api := RendererAPI{router: router}
+
+    handleFunc := func(pattern string, handlerFunc func(http.ResponseWriter, *http.Request)) {
+        // Configure the "http.route" for the HTTP instrumentation.
+        handler := otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc))
+        api.router.Handle(pattern, handler)
+    }
+
+    api.router.StrictSlash(true).Path("/health").HandlerFunc(hc.Handler)
+    handleFunc("/render/{render_type}", api.renderTable)
+    handleFunc("/parse/html", api.parseHTML)
+    return &api
+}
+```
+
+
+The following shows an alternative way to instrument:
+```
+func CreateRendererAPI(ctx context.Context, bindAddr string, allowedOrigins string, errorChan chan error, hc *healthcheck.HealthCheck) {
+    router := mux.NewRouter()
+    routes(router, hc)
+    otelhandler := otelhttp.NewHandler(router, "/")
+
+    httpServer := dphttp.NewServer(bindAddr, otelhandler)
+    // Disable this here to allow main to manage graceful shutdown of the entire app.
+    httpServer.HandleSignals = false
+
+    go func() {
+        log.Info(ctx, "starting table renderer")
+        if err := httpServer.ListenAndServe(); err != nil {
+            errorChan <- err
+            log.Error(ctx, "error occurred when running ListenAndServe", err)
+        }
+    }()
+}
+```
+
+
+A purley middleware approach can also be taken using Alice. Here you can see the instrumentation and logging middlewares applied:
+```
+func New(cfg Config) http.Handler {
+    router := mux.NewRouter()
+
+    middleware := alice.New(
+        dprequest.HandlerRequestID(16),
+        dpotelgo.OtelLoggingMiddleware,
+        otelhttp.NewMiddleware("dp-frontend-router"),
+        log.Middleware,
+        SecurityHandler,
+        healthcheckHandler(cfg.HealthCheckHandler),
+        serverError.Handler,
+        redirects.Handler,
+    )
+
+    newAlice := middleware.Then(router)
+}
+```
+
+
+A slightly different approach has to be taken if using a reverse proxy:
```

Comment:
lines 184-198 can probably be removed as the section "instrumenting http calls" covers it.